commit 2dd2acac897d4fdf201b98f666cfc9bc13cfa33e
Author: Avi Halachmi (:avih) <avihpit@yahoo.com>
Date:   Wed Sep 20 05:50:13 2017 +0300

js: fix broken mp.set_property_number, mp.set_property_native

Also implicitly fixes memory leak when mp.set_property_native was used,
because the cleanup did not expect more allocations from the accidental
use of mpv_get_property.

diff --git a/player/javascript.c b/player/javascript.c
index abb1581f86..14c1872d67 100644
--- a/player/javascript.c
+++ b/player/javascript.c
@@ -616,7 +616,7 @@ static void script_set_property_native(js_State *J, void *af)
     mpv_node node;
     makenode(af, &node, J, 2);
     mpv_handle *h = jclient(J);
-    int e = mpv_get_property(h, js_tostring(J, 1), MPV_FORMAT_NODE, &node);
+    int e = mpv_set_property(h, js_tostring(J, 1), MPV_FORMAT_NODE, &node);
     push_status(J, e);
 }

@@ -647,7 +647,7 @@ static void script_set_property_number(js_State *J)
 {
     double v = js_tonumber(J, 2);
     mpv_handle *h = jclient(J);
-    int e = mpv_get_property(h, js_tostring(J, 1), MPV_FORMAT_DOUBLE, &v);
+    int e = mpv_set_property(h, js_tostring(J, 1), MPV_FORMAT_DOUBLE, &v);
     push_status(J, e);
 }

